components/scream/src/physics/share/physics_constants.hpp

```diff
@@ -109,30 +113,32 @@ struct Constants
   static constexpr int IPARAM         = 3;
 
   // Gases
-  static const std::map<std::string,Scalar> gas_mol_weights;
-  static Scalar get_gas_mol_weight(std::string gas_name);
+  static Scalar get_gas_mol_weight(ci_string gas_name);
 };
 
 // Gases
 // Define the molecular weight for each gas, which can then be
 // used to determine the volume mixing ratio for each gas.
 template <typename Scalar>
-const std::map<std::string,Scalar> Constants<Scalar>::gas_mol_weights = {
-  {"h2o", Scalar(Constants<Scalar>::MWH2O)},
-  {"co2", 44.0095},
-  {"o3" , 47.9982},
-  {"n2o", 44.0128},
-  {"co" , 28.0101},
-  {"ch4", 16.04246},
-  {"o2" , 31.998},
-  {"n2" , 28.0134}
-};
-
-template <typename Scalar>
-Scalar Constants<Scalar>::get_gas_mol_weight(std::string gas_name) {
+Scalar Constants<Scalar>::get_gas_mol_weight(ci_string gas_name) {
   //TODO: Possible improvement would be to design a device friendly function
-  //      here instead of just a std::map lookup, which only works on host.
-  return Constants<Scalar>::gas_mol_weights.at(gas_name);
+  if        (gas_name == "h2o") {
```

Comment:
Quick question for @bartgol , is this formulation already device friendly?  Or at least is there a simple way to make it device friendly?

Reply:
No. `std::string` does not work on device. The easiest way to make a device friendly lookup table is to do

```
enum GasIdx : int {
   idx_h20 = 0,
   idx_co2 = 1,
   ...
   num_gases = 7
};
const Scalar gas_mol_weights = { MWH20, 44.0095, ... };
```
You should then pre-compute the idx of the gases you need in the init phase (where, being on host, you can use string comparisons), and store them in a dev view. Then in the kernels, do something like
```
  auto mw_k = Constants<ScalarT>::gas_mol_weights[my_gases_indices[k]];
```

Reply:
@bartgol , okay cool.  I didn't really think it would be that easy :-)

Given that this PR is mostly focused on the coupling of the physics I would still punt on making get_gas_mol_weights device friendly to a later development.